2022-RADIUSS-AWS/JupyterNotebook/radiuss-aws-flux.ipynb

```diff
@@ -0,0 +1,944 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f0e82702",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!flux mini bulksubmit --watch --wait echo {} ::: foo bar baz"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "392a8056-1661-4b76-9ca3-5e536c687e82",
+   "metadata": {},
+   "source": [
+    "The `--cc` option to `mini submit` makes repeated submission even easier via, `flux mini submit --cc=IDSET`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0ea1962b-1831-4bd2-8dab-c61fd710df9c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!flux mini submit --cc=1-10 --watch hostname"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "27ca3706-8bb4-4fd6-a37c-e6135fb05604",
+   "metadata": {},
+   "source": [
+    "Try it in the JupyterLab terminal with a progress bar and jobs/s rate report: `flux mini submit --cc=1-100 --watch --progress --jps hostname`\n",
+    "\n",
+    "Note that `--wait` is implied by `--watch`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4c5a18ff-8d6a-47e9-a164-931ed1275ef4",
+   "metadata": {},
+   "source": [
+    "Of course, Flux can launch more than just single-node, single-core jobs.  We can submit multiple heterogeneous jobs and Flux will co-schedule the jobs while also ensuring no oversubscription of resources (e.g., cores).\n",
+    "\n",
+    "Note: in this tutorial, we cannot assume that the host you are running on has multiple cores, thus the examples below only vary the number of nodes per job.  Varying the `cores-per-task` is also possible on Flux when the underlying hardware supports it (e.g., a multi-core node)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "brazilian-former",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!flux mini submit --nodes=2 --ntasks=2 --cores-per-task=1 --job-name simulation sleep inf\n",
+    "!flux mini submit --nodes=1 --ntasks=1 --cores-per-task=1 --job-name analysis sleep inf"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3f8c2af2",
+   "metadata": {},
+   "source": [
+    "### Listing job properties with `flux jobs`\n",
+    "\n",
+    "We can now list out the jobs in the queue with `flux jobs`, and we should see both jobs that we just submitted."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "institutional-vocabulary",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!flux jobs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "77ca4277",
+   "metadata": {},
+   "source": [
+    "Since those jobs won't ever exit (and we didn't specify a timelimit), let's kill them off now and free up the resources."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "46dd8ec8-6c64-4d8d-9a00-949f5f58c07b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!flux job killall -f\n",
+    "!flux jobs"
```

Comment:
We should add an example of listing jobs that have completed, been canceled or failed as well with `flux jobs -a`. Note that in the terminal, jobs completed successfully will be highlighted in green, and those that failed in red.

Reply:
Oh, I now see you have that below (sorry going through this all at once)